File: Projects/Dagger/Projects/Project-4-Qualifiers/app/src/main/java/com/droid/code/ui/MainViewModel.java
package com.droid.code.ui;

import com.droid.code.data.local.DatabaseService;
import com.droid.code.data.remote.NetworkService;

import javax.inject.Inject;

public class MainViewModel {

    private DatabaseService databaseService;
    private NetworkService networkService;

    @Inject
    public MainViewModel(DatabaseService databaseService, NetworkService networkService) {
        this.databaseService = databaseService;
        this.networkService = networkService;
    }

    public String getSomeData() {
        return databaseService.getDummyData() + " : " + networkService.getDummyData();
    }
}

File: Projects/Dagger/Projects/Project-5-DaggerInKotlin/app/src/main/java/com/demo/kotlin/sample/ui/main/MainViewModel.kt
package com.demo.kotlin.sample.ui.main

import com.demo.kotlin.sample.data.local.DatabaseService
import com.demo.kotlin.sample.data.remote.NetworkService
import com.demo.kotlin.sample.di.ActivityScope

import javax.inject.Inject

@ActivityScope
class MainViewModel @Inject constructor(
        private val databaseService: DatabaseService,
        private val networkService: NetworkService) {

    val someData: String
        get() = databaseService.dummyData + " : " + networkService.dummyData
}
